### PR TITLE
Add unit tests for the account closure feature

### LINKED ISSUE #18456: Add unit tests
This task is to add unit tests for the account closure feature.

### PR DESCRIPTION
**Pull request by mkevins**
Fixes #18456

To test:

Verify that the unit tests pass on CI.

## Regression Notes
1. Potential unintended areas of impact
Other nearby unit tests.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Existing automated tests.

3. What automated tests I added (or what prevented me from doing so)
This PR adds unit tests for the account closure feature.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/test/java/org/wordpress/android/ui/prefs/accountsettings/AccountSettingsViewModelTest.kt [hunk 1]
```diff
@@ -443,10 +476,19 @@ class AccountSettingsViewModelTest : BaseUnitTest() {
...
-            mock(),
+            accountClosureUseCase,
         )
     }
 
+    private suspend fun mockAccountClosureWithResult(result: CloseAccountResult) {
+        whenever(getSitesUseCase.getAtomic()).thenReturn(emptyList())
+        whenever(accountClosureUseCase.closeAccount(any())).thenAnswer {
+            val completion = it.getArgument<((CloseAccountResult) -> Unit)>(0)
+            completion.invoke(result)
```
Comment (ovitrif) [inline WordPress/src/test/java/org/wordpress/android/ui/prefs/accountsettings/AccountSettingsViewModelTest.kt:487]: First time I see something like this, neat 👌🏻

**Review by ovitrif — APPROVED** [review]
LGTM 🎉 

Thank you for adding unit tests @mkevins 🏆 

There's some lint failures surfaced by the CI jobs, feel free to merge once those are resolved 👍🏻

**Review by ovitrif — APPROVED** [review]
LGTM 🎉 

Thank you for adding unit tests @mkevins 🏆 

There's some lint failures surfaced by the CI jobs, feel free to merge once those are resolved 👍🏻

**Review by antonis — APPROVED** [review]
Great work @mkevins 🎉 
The added unit test look great. Thank you for your work on this 🙇 

I took the liberty of removing an extra line to fix a lint check with bdf5833a2fff66ea97ff7170228a921b979d2b1d

**Comment by mkevins** [comment]
Thank you both for reviewing, and thanks Antonis for fixing the empty line to fix the lint!

**Pull request merged**